src/components/Facets.tsx

```diff
@@ -29,17 +26,95 @@ export interface FacetsProps {
  * and {@link HierarchicalFacets} components can be used instead for more control over facet
  * configuration.
  *
+ * To override a single facet, use {@link StandardFacet}.
+ *
  * @param props - {@link FacetsProps}
  * @returns A React component for facets
  *
  * @public
  */
 export function Facets(props: FacetsProps) {
-  const { customCssClasses = {} } = props;
+  const { searchOnChange, children, customCssClasses = {} } = props;
+
+  const fieldIdToCustomFacetProps = new Map();
+  if (children) {
+    (Array.isArray(children) ? children : [children])
+      .forEach(child => fieldIdToCustomFacetProps.set(child.props.fieldId, child));
+  }
+
   return (
-    <div className={customCssClasses.facetsContainer}>
-      <StandardFacets/>
-      <NumericalFacets />
+    <div>
+      <FacetsProvider searchOnChange={searchOnChange} className={customCssClasses.facetsContainer}>
+        {facets => facets
+          .map((facet, i) => {
+            let facetType: FacetType = FacetType.STANDARD;
+            let facetProps: FacetProps = {
+              fieldId: facet.fieldId,
+              label: facet.displayName,
+            };
+            if (fieldIdToCustomFacetProps.has(facet.fieldId)) {
+              const customFacetElement: ReactElement = fieldIdToCustomFacetProps.get(facet.fieldId);
+              facetProps = customFacetElement.props;
+              facetType = getFacetTypeFromReactElementType(
+                (typeof customFacetElement.type === 'function') ? customFacetElement.type.name : '');
+            } else {
+              if (isStringFacet(facet)) {
+                facetType = FacetType.STANDARD;
+              } else if (isNumericalFacet(facet)) {
+                facetType = FacetType.NUMERICAL;
+              }
+            }
+
+            let facetComponent: ReactElement;
+            switch (facetType) {
+              case FacetType.NUMERICAL:
+                facetComponent = <div/>;
+                break;
+              case FacetType.STANDARD:
+                // fall through
+              default:
+                facetComponent = (
+                  <StandardFacetContent
+                    facet={facet}
+                    {...facetProps}
+                  />);
+            }
+
+            return (
+              <Fragment key={facet.fieldId}>
+                {facetComponent}
+                {(i < facets.length - 1) && <FilterDivider className={customCssClasses?.divider}/>}
+              </Fragment>
+            );
+          })
+        }
+      </FacetsProvider>
+      <NumericalFacets/>
     </div>
   );
-}
\ No newline at end of file
+}
+
+/**
+ * A component that displays a single standard facet. Use this to override the default rendering.
+ *
+ * @param props - {@link StandardFacetProps}
+ * @returns ReactElement
+ * @public
+ */
+export function StandardFacet(props: StandardFacetProps) { return null; }
```

Comment:
The general convention in this library is for each public component to be in its own file, along with the props and css classes in that file as well. Is there a reason behind this organization instead?

Reply:
I put it here since in the [spec](https://docs.google.com/document/d/1V1n_mqzZQATM4uQVcmj0_Ia8e-bYXDG58t7G9t0Lop8/edit#), they use `<Facets.StandardFacet/>`. Let me know what you think!